python/lsst/ts/standardscripts/maintel/take_image_anycam.py

```diff
@@ -0,0 +1,448 @@
+        if "comcam" in schema_dict and "filter" not in schema_dict["comcam"]:
+            schema_dict["comcam"]["filter"] = None
+
+        if "lsstcam" in schema_dict and "filter" not in schema_dict["lsstcam"]:
+            schema_dict["lsstcam"]["filter"] = None
+
+        base_schema_dict = super().get_schema()
+
+        for properties in base_schema_dict["properties"]:
+            schema_dict["properties"][properties] = base_schema_dict["properties"][
+                properties
+            ]
+
+        return schema_dict
+
+    async def configure(self, config):
+        """Configure the script based on the provided configuration.
+
+        This method initializes the MTCS and any cameras specified in the
+        configuration (`config`). It supports initializing ComCam, LSSTCam,
+        and an arbitrary number of Generic Cameras.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration namespace containing camera-specific configurations.
+
+        Notes
+        -----
+        - `self.mtcs`: An instance of the MTCS. Initialized
+        only once and used to control telescope components beyond cameras.
+
+        - `self.comcam`, `self.lsstcam`: Direct references to initialized
+          ComCam and LSSTCam instances, respectively. These attributes allow
+          for direct interaction with the camera instances for operations
+          not necessarily tied to a specific imaging configuration (e.g.,
+          status checks, hardware interactions).
+
+        - `self.generic_cams`: A list of initialized GenericCamera instances.
+          Similar to `self.comcam` and `self.lsstcam`, it allows for direct
+          interactions with each generic camera. This list is populated only
+          if generic cameras are specified in the configuration.
+
+        - `self.camera_setups`: A list of `CameraSetup`
+          instances. Each instance encapsulates a camera object (ComCam,
+          LSSTCam, or GenericCamera), its specific imaging configuration,
+          and an identifier. This list is the primary resource for executing
+          configured imaging operations across all cameras, as it provides
+          a unified interface to interact with cameras based on their
+          configurations.
+
+        The method ensures that MTCS and each camera are initialized only once.
+        For generic cameras, initialization tasks are executed concurrently to
+        optimize startup time.
+        """
+
+        self.config = config
+
+        # Initialize MTCS
+        if self.mtcs is None:
+            self.log.debug("Creating MTCS.")
+            self.mtcs = MTCS(domain=self.domain, log=self.log)
+            await self.mtcs.start_task
+        else:
+            self.log.debug("MTCS already defined, skipping.")
+
+        self.camera_setups = []
+
+        # Initialize LSSTCam if configured
+        if self.lsstcam is None and hasattr(config, "lsstcam"):
```

Comment:
If you adopt my suggestion to remove these references and only hold a dict of `CameraSetup`, you could do something like this instead:

```
if "lsstcam" not in self.camera_setups and hasattr(config, "lsstcam"):
    self.log.debug("Creating LSSTCam.")
    lsstcam = LSSTCam(self.domain, log=self.log)
    await lsstcam.start_task
    self.camera_setups["lsstcam"] = CameraSetup(self.lsstcam, config.lsstcam, "LSSTCam")
```

Then do the same with "comcam" and "gencam".